src/cluster.py

```diff
@@ -766,6 +770,13 @@ def switchover(self, candidate: str | None = None) -> None:
 
         # Check whether the switchover was unsuccessful.
         if r.status_code != 200:
+            if (
+                r.status_code == 412
+                and r.text == "candidate name does not match with sync_standby"
+            ):
+                logger.debug("Unit is not sync standby")
+                raise SwitchoverNotSyncError()
```

Comment:
This will happen when trying to promote the current primary or the async replica.